model/span.go

```diff
@@ -34,12 +34,22 @@ const (
 	FirehoseFlag = Flags(8)
 
 	samplerType        = "sampler.type"
+	keySpanKind        = "span.kind"
 	samplerTypeUnknown = "unknown"
 )
 
 // Flags is a bit map of flags for a span
 type Flags uint32
 
+// Map from string to trace.SpanKind
+var spanKindMapping = map[string]trace.SpanKind{
```

Comment:
nit: I tend to prefer avoid package-level vars where practical, especially in this case where the lookup should never change, because as a package-level var, it can be mutated throughout the lifetime of the program.

What do you think about making this readonly?

```go
// Map from string to trace.SpanKind.
func toSpanKind(s string) (trace.SpanKind, bool) {
	sk, found := map[string]trace.SpanKind{
		"client":   trace.SpanKindClient,
		"server":   trace.SpanKindServer,
		"producer": trace.SpanKindProducer,
		"consumer": trace.SpanKindConsumer,
		"internal": trace.SpanKindInternal,
	}[s]
	return sk, found
}
```

Reply:
would this allocate a map on every invocation?

Reply:
Potentially, but worst case, I expect it to be very cheap compared to I/O, for instance; but hard to know the actual impact without profiling. If we think this will be on a hot path and materially impact performance, I'm happy to leave it as is.

Reply:
By the way, since this is a variable declared inside the stack frame of this function call, I expect it to be allocated on the stack, which should be substantially more efficient than heap allocation.

Reply:
thanks for that pro tip. I discovered new way of doing things sorted.

Reply:
it's not just the matter of allocating, actually initializing a map and laying out its structure is unnecessary work. It's normal to keep private maps as package var.